lesson_2/mortgage_calc.py

```diff
@@ -0,0 +1,158 @@
+        prompt(MESSAGES["years"])
+        year_duration = input()
+    year_month_conversion = int(year_duration) * 12
+    return year_month_conversion
+
+def get_loan_month_duration():
+    prompt(MESSAGES["months"])
+    month_duration = input()
+    while invalid_loan_duration(month_duration):
+        error_output(MESSAGES["invalid_months"])
+        prompt(MESSAGES["months"])
+        month_duration = input()
+    month_duration = int(month_duration)
+    return month_duration
+
+def invalid_loan_duration(loan_duration):
+    try:
+        int(loan_duration)
+        if int(loan_duration) < 0:
+            return True
+    except ValueError:
+        return True
+    return False
+
+def zero_or_less_duration(year_month_conversion, month_duration):
+    duration = year_month_conversion + month_duration
+    if duration <= 0:
+        return True
+    return False
+
+def calc_again():
+    prompt(MESSAGES["another_calculation"])
+    another_calc = input()
+    if another_calc.lower() == 'y':
+        print('\n-------NEW CALCULATION-------')
+        return loan_calculator()
+
+    return intro_outro('SEE YOU LATER!')
+
+# LOAN CALC PROGRAM - uses functions above
+def loan_calculator():
+    # LOAN AMOUNT INPUT:
+    loan_amount_input = get_loan_amount()
+    while invalid_loan_amount(loan_amount_input):
+        error_output(MESSAGES["invalid_loan_amount"])
+        loan_amount_input = get_loan_amount()
+    loan_amount = float(loan_amount_input)
+
+    # MONTHLY APR:
+    annual_percentage_rate_input = get_apr()
+    while invalid_apr(annual_percentage_rate_input):
+        error_output(MESSAGES["invalid_apr"])
+        annual_percentage_rate_input = get_apr()
+    annual_percentage_rate = float(annual_percentage_rate_input)
+
+    if annual_percentage_rate != 0:
+        # CONVERT APR TO DECIMAL FORMAT
+        apr_decimal = float(annual_percentage_rate_input) * .01
+        # APR -> MONTHLY INTEREST RATE
+        monthly_interest_rate = apr_decimal / 12
+    else:
+        annual_percentage_rate = 'No APR'
+
+    # LOAN DURATION:
+    prompt(MESSAGES["loan_duration"])
+    year_month_conversion = get_loan_year_duration()
+    month_duration = get_loan_month_duration()
+
+    # CHECK TOTAL LOAN DURATION IS GREATER THAN 0:
+    while zero_or_less_duration(year_month_conversion, month_duration):
+        error_output(MESSAGES["invalid_loan_duration"])
+        year_month_conversion = get_loan_year_duration()
+        month_duration = get_loan_month_duration()
+
+    loan_duration_months = year_month_conversion + month_duration
+
+    # MONTHLY PAYMENT with VS without APR:
+    if annual_percentage_rate != 'No APR':
+        total_monthly_payment = loan_amount * (monthly_interest_rate /
+        (1 - (1 + monthly_interest_rate) ** (-loan_duration_months)))
+        total_duration_interest = (
+        (total_monthly_payment * loan_duration_months)- loan_amount
+        )
+        total_payment = loan_amount + total_duration_interest
```

Comment:
This would be a good candidate for a function and you can document what it's doing with a comment for future developers:

example:
```python
"""
Calculates the total monthly payment, total interest paid, and total payment for a loan.

Parameters:
    - loan_amount (float): The initial loan amount
    - monthly_interest_rate (float): The monthly interest rate (as a decimal)
    - loan_duration_months (int): The total duration of the loan in months

Returns:
    - tuple: (total_monthly_payment, total_duration_interest, total_payment)
"""
def calculate_loan(loan_amount, monthly_interest_rate, loan_duration_months):
    total_monthly_payment = loan_amount * (monthly_interest_rate / 
                                          (1 - (1 + monthly_interest_rate) ** (-loan_duration_months)))
    total_duration_interest = (total_monthly_payment * loan_duration_months) - loan_amount
    total_payment = loan_amount + total_duration_interest
    
    return total_monthly_payment, total_duration_interest, total_payment

total_monthly_payment, total_duration_interest, total_payment = calculate_loan(10000, 0.05/12, 60)
```